### PR TITLE
Add --verify option to spead2_send and spead2_recv

### PR DESCRIPTION
**Pull request by bmerry**
It puts random data (created from a linear congruential generator) into
the data stream, and verifies it on the receive side. This can help with
testing issues that are missed by unit tests e.g. corner cases that only
affect a small number of heaps, or that require the buffer to be
recycled before they show up.

### CODE REVIEW

**Review by gcallanan — COMMENTED** [review]
I am just a bit unclear on how this all fits together. Left a comment asking for clarity.

File: src/spead2_recv.cpp [hunk 1]
```diff
@@ -170,6 +174,82 @@ void show_heap(const spead2::recv::heap &fheap, const options &opts)
     std::cout << std::noshowbase;
 }
 
+/* O(log(z)) version of engine.discard(z), which in GCC seems to be implemented in O(z). */
```
[the remark this answers is not in the document]
Comment (bmerry) [reply src/spead2_recv.cpp:177]: Does https://en.cppreference.com/w/cpp/numeric/random/linear_congruential_engine/discard help? It's jumping forward z steps in the random sequence.

**Review by gcallanan — APPROVED** [review]
